### DISCUSSION TITLE
withdrawStart in POS gives error

### DISCUSSION THREAD

**Discussion by m-a-x-i-m-u-s** [0xPolygon/matic.js#328] [Q&A] [answered]
While calling withdrawStart with POSClient an error occurs due to the sender address being undefined. 
A solution to the problem is specifying the sender address in the "from" field of second parameter of the withdrawStart function.
![image](https://user-images.githubusercontent.com/85619385/161343276-c90229ad-e7c3-4914-8bd8-1e9da8ce7dad.png)

**Comment by ujjwalgupta94** [comment, marked as answer]
this is because you have not provided `from`  value. You can provide `from` in transaction option or you can provide default value in client.

```
const result = await withdrawStart(100,{
   from:'from address'
})
```

or in the client as default from - 

```
 const posClient = new POSClient()
  return posClient.init({
    log: true,
    network: network,
    version: version,
    child: {
      provider: new HDWalletProvider(privateKey, config.child.rpc),
      defaultConfig: {
        from: userAddress
      }
    },
    parent: {
      provider: new HDWalletProvider(privateKey, config.parent.rpc),
      defaultConfig: {
        from: userAddress
      }
    }
  });
```

this one is more recommended, as it will save you to provide value in each transaction.
